pandas/core/frame.py

```diff
@@ -8190,11 +8190,16 @@ def update(
         if not isinstance(other, DataFrame):
             other = DataFrame(other)
 
-        other = other.reindex_like(self)
+        # reindex rows, non-matching columns get skipped
+        other = other.reindex(self.index)
 
         for col in self.columns:
+            if col not in other.columns:
```

Comment:
Could probably do a union of `self.columns.union(other.columns)` first and iterate over that instead of constantly doing `in`

Reply:
This makes a lot of sense, thank you. I think it needs to be intersection though right? I added a new commit converting to sets to do intersection.

Reply:
Ah yes, thanks for the correction: yes it should be the intersection.

Also `self.columns` should have an `intersection` method that can be used like `self.columns.intersection(other.columns)`

Reply:
Ahh I see, switched to using the built in intersection